plugin/restore/mysql/mysql.go

```diff
@@ -457,92 +492,193 @@ func getPITROldDatabaseName(database string, suffixTs int64) string {
 	return getSafeName(database, suffix)
 }
 
-// FetchArchivedBinlogFiles downloads the missing binlog files from the remote instance to `binlogDir`,
-// but exclude latest binlog. We may  download the latest binlog only when doing PITR.
-func (r *Restore) FetchArchivedBinlogFiles(ctx context.Context) error {
-	binlogFilesLocal, err := ioutil.ReadDir(r.binlogDir)
-	if err != nil {
-		return err
+// Parse the first eventTs of a local binlog file.
+func (r *Restore) parseFirstLocalBinlogEventTimestamp(ctx context.Context, fileName string) (int64, error) {
+	// https://dev.mysql.com/doc/internals/en/binlog-file-header.html
+	// > A binlog file starts with a Binlog File Header \0xfe'bin'
+	// The starting point of the first binlog event is at position 4 of a binlog file.
+	firstBinlogEvent := api.BinlogInfo{FileName: fileName, Position: 4}
+	return r.parseLocalBinlogEventTimestamp(ctx, firstBinlogEvent)
+}
+
+func convertToBinlogFiles(binlogFileInfoList []fs.FileInfo) ([]BinlogFile, error) {
+	var binlogFileList []BinlogFile
+	for _, fileInfo := range binlogFileInfoList {
+		binlogFile, err := newBinlogFile(fileInfo.Name(), fileInfo.Size())
+		if err != nil {
+			return nil, err
+		}
+		binlogFileList = append(binlogFileList, binlogFile)
 	}
+	return binlogFileList, nil
+}
 
-	binlogFilesOnServer, err := r.getBinlogFilesMetaOnServer(ctx)
+func getLocalBinlogFiles(binlogDir string) ([]BinlogFile, error) {
+	binlogFilesInfoLocal, err := ioutil.ReadDir(binlogDir)
 	if err != nil {
-		return err
+		return nil, err
 	}
-
-	latestBinlogFileOnServer, err := r.getLatestBinlogFileMeta(ctx)
+	binlogFilesLocal, err := convertToBinlogFiles(binlogFilesInfoLocal)
 	if err != nil {
-		return err
+		return nil, err
 	}
+	binlogFilesLocalSorted, err := sortBinlogFiles(binlogFilesLocal)
+	if err != nil {
+		return nil, err
+	}
+	return binlogFilesLocalSorted, nil
+}
 
-	// build a local file size map from file name to size
-	localFileMap := make(map[string]int64)
+func localBinlogFilesAreContinuous(files []BinlogFile) bool {
```

Comment:
if it's called localBinlogFilesAreContinuous, then it's more appropriate to pass the binlogDir like getLocalBinlogFiles.

Or we just call it binlogFilesAreContinuous

Reply:
Fixed.